pkg/services/ngalert/remote/client/alertmanager_state.go

```diff
@@ -17,7 +18,15 @@ type UserState struct {
 	State string `json:"state"`
 }
 
+func (mc *Mimir) GetFullState(ctx context.Context) (*UserState, error) {
+	return mc.getState(ctx, fullStatePath)
+}
+
 func (mc *Mimir) GetGrafanaAlertmanagerState(ctx context.Context) (*UserState, error) {
+	return mc.getState(ctx, grafanaAlertmanagerStatePath)
```

Comment:
Why do we need two "state" APIs? 
- GetGrafanaAlertmanagerState seems to be used in tests only. The implementation of it on the server https://github.com/grafana/mimir/blob/92ec32b67caa45eee9c521523f6bb0afb05a067c/pkg/alertmanager/api_grafana.go#L182-L192 also returns full state.
- Judging by name GetFullState does not mention Grafana and it could mean it returns something else than Grafana Alertmanager's state.

Reply:
There are two different states we can fetch from Mimir:
1. Grafana state: sent from Grafana, stored in object storage, not yet merged with Mimir's state
2. Mimir's full state: the Alertmanager's internal state, not necessarily containing state sent from Grafana

The endpoint you mention gets the state from object storage, not the running Alertmanager. This will return a 404 after the configuration is promoted, as Mimir deletes this state after merging it with its own.

What we're getting here is Mimir's state. It can contain Grafana state, but not necessarily (e.g. if the Alertmanager has been running long enough for all Grafana nflog entries and silences to be GC'd).